kf_lib_data_ingest/common/file_retriever.py

```diff
@@ -84,52 +84,50 @@ def _s3_save(protocol, source_loc, dest_obj, auth=None, logger=None):
     raise botocore.exceptions.NoCredentialsError()  # never got the file
 
 
-def _web_save(protocol, source_loc, dest_dir, delete, auth=None, logger=None):
+def _web_save(protocol, source_loc, dest_obj, auth=None, logger=None):
     """
     Get contents of a file from a web server to a local file-like object.
 
-    Preserves the file extension of the original file if Content-Disposition
-    header is provided in request.
+    Preserves the name of the original file if Content-Disposition
+    header is provided in the response.
     See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition # noqa E501
 
     :param protocol: URL protocol identifier
     :type protocol: str
     :param source_loc: address or path
     :type source_loc: str
-    :param dest_dir: Directory where temp file will be downloaded
-    :type dest_dir: str
-    :param delete: Flag indicating whether or not to delete file after download
-    :type delete: bool
+    :param dest_obj: Receives the data downloaded from the source file
+    :type dest_obj: File-like object
     :param auth: optional requests-compatible auth object
     :type auth: http://docs.python-requests.org/en/master/user/authentication/
 
-    :returns: the file-like object containing the downloaded data
+    :returns: None, the data goes to dest_obj
     """
     logger = logger or logging.getLogger(__name__)
     url = protocol + PROTOCOL_SEP + source_loc
-    dest_obj = None
     with requests.get(url, auth=auth, stream=True) as response:
         if response.status_code == 200:
-            # Get filename and extension from Content-Disposition header
-            file_ext = None
+            # Get filename from Content-Disposition header
             content_disposition = response.headers.get('Content-Disposition',
                                                        '')
-            parts = content_disposition.split(';')
+            _, cdisp_params = cgi.parse_header(content_disposition)
+            filename = cdisp_params.get('filename*')
+            # RFC 5987 ext-parameter is actually more complicated than this,
+            # but this should get us 90% there, and the rfc6266 python lib is
+            # broken after PEP 479. *sigh* - Avi K
+            if filename and filename.lower.startswith('utf-8'):
+                filename = filename.split("'", 2)[2].decode('utf-8')
+            else:
+                filename = cdisp_params.get('filename')
 
             # Header did not provide filename
-            if len(parts) < 2:
+            if filename:
+                dest_obj.original_name = filename
```

Comment:
To be consistent with what gets assigned to original name using other getters:
```suggestion
                dest_obj.original_name = source_loc
```